### ISSUE TITLE
Timeout is not working correctly with parallel=tests when xmltest contains multiple method

### ISSUE THREAD

**Issue by tunefun** [cbeust/testng#2482]
### TestNG Version 
7.3.0

### Expected behavior
Test timeout should be total time of all method but not each

### Actual behavior
Test timeout act as method timeout

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample
```java
import org.testng.annotations.Test;

import java.util.concurrent.TimeUnit;

public class Issue779TestcaseSample1 {
    @Test
    public void testMethod1() throws InterruptedException {
        TimeUnit.SECONDS.sleep(5);
    }

    @Test
    public void testMethod2() throws InterruptedException {
        TimeUnit.SECONDS.sleep(5);
    }

    @Test
    public void testMethod3() throws InterruptedException {
        TimeUnit.SECONDS.sleep(10);
    }
}
```

**Test runner**

```java
import org.testng.Assert;
import org.testng.TestListenerAdapter;
import org.testng.TestNG;
import org.testng.xml.XmlClass;
import org.testng.xml.XmlSuite;
import org.testng.xml.XmlTest;

import java.util.Collections;

public class Issue779TestRunner {
    public static void main(String[] args) {
        XmlSuite suite = new XmlSuite();
        suite.setTimeOut(String.valueOf(60 * 1000));
        suite.setParallel(XmlSuite.ParallelMode.TESTS);
        suite.setName("suite");

        XmlTest xmlTest = new XmlTest(suite);
        xmlTest.setTimeOut(7 * 1000);
        xmlTest.setXmlClasses(Collections.singletonList(new XmlClass(Issue779TestcaseSample1.class)));
        xmlTest.setName("test");

        TestNG testng = new TestNG();
        testng.setXmlSuites(Collections.singletonList(suite));
        testng.setVerbose(2);
        TestListenerAdapter tla = new TestListenerAdapter();
        testng.addListener(tla);
        testng.run();

        Assert.assertEquals(tla.getPassedTests().size(), 1, "pass methods count");
    }
}

```

**Comment by krmahadevan** [comment]
@tunefun - I think you are understanding this incorrectly.

Quoting the documentation from the `TestNG.dtd` https://github.com/cbeust/testng/blob/f9fcf66f3631f07b27b72be032c352978a4e7c6e/src/main/resources/testng-1.0.dtd#L54-L55

```
the time to wait in milliseconds before aborting the method (if parallel="methods") or the test (if parallel="tests")
```

This parameter works at the **suite** level  ( `<suite>` ) for the below two parallel execution strategies:

* tests
* methods

This parameter works at the **test** level ( `<test>` ) when the parallel execution strategy is `methods`.

In your case, since you have just added one `<test>` tag, and the parallelism strategy is `tests`. Since there's only one `<test>` tag, the methods in this `<test>` tag are going to run sequentially and TestNG is going to apply a timeout value of `7000` milliseconds for each of these methods.

Per your sample, only `testMethod3()` will fail (because it sleeps for 10 seconds) and the other two will pass.

So this assertion is incorrect

```
Assert.assertEquals(tla.getPassedTests().size(), 1, "pass methods count");
```

In a nutshell, the `time-out` attribute is always intended to be measured at particular method level and becomes applicable at `<suite>` level (when parallel = `tests|methods`) or at `<test>` level (when parallel='methods') 

Closing this issue with resolution as **working as designed**.

**Comment by tunefun** [comment]
@krmahadevan How can I make it working as whole test timeout? When I set the suit timeout to 7000 millseconds, this test ends within 7000 millseconds.

**Test runner**

```java
import org.testng.Assert;
import org.testng.TestListenerAdapter;
import org.testng.TestNG;
import org.testng.xml.XmlClass;
import org.testng.xml.XmlSuite;
import org.testng.xml.XmlTest;

import java.util.Collections;

public class Issue779TestRunner {
    public static void main(String[] args) {
        XmlSuite suite = new XmlSuite();
        suite.setTimeOut(String.valueOf(7 * 1000));
        suite.setParallel(XmlSuite.ParallelMode.TESTS);
        suite.setName("suite");

        XmlTest xmlTest = new XmlTest(suite);
        xmlTest.setTimeOut(7 * 1000);
        xmlTest.setXmlClasses(Collections.singletonList(new XmlClass(Issue779TestcaseSample1.class)));
        xmlTest.setName("test");

        TestNG testng = new TestNG();
        testng.setXmlSuites(Collections.singletonList(suite));
        testng.setVerbose(2);
        TestListenerAdapter tla = new TestListenerAdapter();
        testng.addListener(tla);
        testng.run();

        Assert.assertEquals(tla.getPassedTests().size(), 1, "pass methods count");
    }
}

```

**Comment by krmahadevan** [comment]
@tunefun 

> How can I make it working as whole test timeout?

When you enable parallelism as `tests` at `<suite>` level:

* if you have only one `<test>` tag, then you would need to explicitly set `parallel=methods|instances|classes` at the `<test>` level, for TestNG to kick off parallel execution. If you don't do this, then TestNG will treat each and every test method as an individual execution and attempt at running them in separate executorservice
* If you have more than one `<test>` tags, then you don't need to do anything extra.

The `time-out` defined at the `<suite>` level (or) `<test>` level are essentially being passed on to the underlying executor service.
